src/configuration-requestor.ts

```diff
@@ -1,150 +1,121 @@
-    ) {
-      // TODO: Notify that config updated.
+  private async fetchPrecomputedConfiguration(
+    precomputed: NonNullable<ConfigurationRequestorOptions['precomputed']>,
+  ): Promise<Configuration> {
+    const response = await this.httpClient.getPrecomputedFlags(precomputed);
+    if (!response) {
+      throw new ConfigurationError('empty response');
     }
-  }
 
-  private getLoadedBanditModelVersions(entries: Record<string, BanditParameters>): string[] {
-    return Object.values(entries).map((banditParam: BanditParameters) => banditParam.modelVersion);
+    return Configuration.fromResponses({ precomputed: response });
   }
 
-  private requiresBanditModelConfigurationStoreUpdate(
-    currentBanditModelVersions: string[],
-    banditReferences: Record<string, BanditReference>,
-  ): boolean {
-    const referencedModelVersions = Object.values(banditReferences).map(
-      (banditReference: BanditReference) => banditReference.modelVersion,
-    );
-
-    return !referencedModelVersions.every((modelVersion) =>
-      currentBanditModelVersions.includes(modelVersion),
-    );
-  }
+  /**
+   * Get bandits configuration matching the flags configuration.
+   *
+   * This function does not fetch bandits if the client does not want
+   * them (`ConfigurationRequestorOptions.wantsBandits === false`) or
+   * if we can reuse bandit models from `ConfigurationStore`.
+   */
+  private async getBanditsFor(flags: FlagsConfig): Promise<BanditsConfig | undefined> {
+    const needsBandits =
+      this.options.wantsBandits && Object.keys(flags.response.banditReferences ?? {}).length > 0;
+    if (!needsBandits) {
+      return undefined;
+    }
 
-  private indexBanditVariationsByFlagKey(
-    banditVariationsByBanditKey: Record<string, BanditReference>,
-  ): Record<string, BanditVariation[]> {
-    const banditVariationsByFlagKey: Record<string, BanditVariation[]> = {};
-    Object.values(banditVariationsByBanditKey).forEach((banditReference) => {
-      banditReference.flagVariations.forEach((banditVariation) => {
-        let banditVariations = banditVariationsByFlagKey[banditVariation.flagKey];
-        if (!banditVariations) {
-          banditVariations = [];
-          banditVariationsByFlagKey[banditVariation.flagKey] = banditVariations;
-        }
-        banditVariations.push(banditVariation);
-      });
-    });
-    return banditVariationsByFlagKey;
+    const prevBandits = this.latestConfiguration?.getBanditConfiguration();
+    const canReuseBandits = banditsUpToDate(flags, prevBandits);
+    if (canReuseBandits) {
+      return prevBandits;
+    }
+
+    return await this.httpClient.getBanditParameters();
   }
 }
+
+/**
+ * Checks that bandits configuration matches the flags
+ * configuration. This is done by checking that bandits configuration
+ * has proper versions for all bandits references in flags
+ * configuration.
+ */
+const banditsUpToDate = (flags: FlagsConfig, bandits: BanditsConfig | undefined): boolean => {
```

Comment:
💯 I like the inversion here over what we've coded this as (needToFetchBandits). This makes it much more clear.